pkg/reconciler/buildrun/resources/results_test.go

```diff
@@ -0,0 +1,147 @@
+	var ctl test.Catalog
+
+	Context("when a BuildRun complete successfully", func() {
+		var (
+			taskRunRequest reconcile.Request
+			br             *build.BuildRun
+			tr             *pipelinev1beta1.TaskRun
+		)
+
+		ctx := context.Background()
+
+		// returns a reconcile.Request based on an resource name and namespace
+		newReconcileRequest := func(name string, ns string) reconcile.Request {
+			return reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      name,
+					Namespace: ns,
+				},
+			}
+		}
+
+		BeforeEach(func() {
+			taskRunRequest = newReconcileRequest("foo-p8nts", "bar")
+			tr = ctl.DefaultTaskRun("foo-p8nts", "bar")
+			br = ctl.DefaultBuildRun("foo", "bar")
+		})
+
+		It("should surface the TaskRun results emitting from default(git) source step", func() {
+			commitSha := "0e0583421a5e4bf562ffe33f3651e16ba0c78591"
+			br.Status.BuildSpec.Source.URL = "https://github.com/shipwright-io/sample-go"
+
+			tr.Status.TaskRunResults = append(tr.Status.TaskRunResults,
+				pipelinev1beta1.TaskRunResult{
+					Name:  "shp-source-default-commit-sha",
+					Value: commitSha,
+				},
+				pipelinev1beta1.TaskRunResult{
+					Name:  "shp-source-default-commit-author",
+					Value: "foo bar",
+				})
+
+			resources.UpdateBuildRunUsingTaskResults(ctx, br, tr.Status.TaskRunResults, taskRunRequest)
+
+			Expect(len(br.Status.Sources)).To(Equal(1))
+			Expect(br.Status.Sources[0].Git.CommitSha).To(Equal(commitSha))
+			Expect(br.Status.Sources[0].Git.CommitAuthor).To(Equal("foo bar"))
+		})
+
+		It("should surface the TaskRun results emitting from default(bundle) source step", func() {
+			bundleImageDigest := "sha256:fe1b73cd25ac3f11dec752755e2"
+			br.Status.BuildSpec.Source.BundleContainer = &build.BundleContainer{
+				Image: "quay.io/shipwright/source-bundle:latest",
+			}
+
+			tr.Status.TaskRunResults = append(tr.Status.TaskRunResults,
+				pipelinev1beta1.TaskRunResult{
+					Name:  "shp-source-default-image-digest",
+					Value: bundleImageDigest,
+				})
+
+			resources.UpdateBuildRunUsingTaskResults(ctx, br, tr.Status.TaskRunResults, taskRunRequest)
+
+			Expect(len(br.Status.Sources)).To(Equal(1))
+			Expect(br.Status.Sources[0].Bundle.Digest).To(Equal(bundleImageDigest))
+		})
+
+		It("should surface the TaskRun results emitting from output step", func() {
+			imageDigest := "sha256:fe1b73cd25ac3f11dec752755e2"
+
+			tr.Status.TaskRunResults = append(tr.Status.TaskRunResults,
+				pipelinev1beta1.TaskRunResult{
+					Name:  "shp-image-digest",
+					Value: imageDigest,
+				},
+				pipelinev1beta1.TaskRunResult{
+					Name:  "shp-image-size",
+					Value: "230",
+				})
+
+			resources.UpdateBuildRunUsingTaskResults(ctx, br, tr.Status.TaskRunResults, taskRunRequest)
+
+			size, err := strconv.ParseInt("230", 10, 64)
+			Expect(err).To(BeNil())
+
+			Expect(br.Status.Output.Digest).To(Equal(imageDigest))
+			Expect(br.Status.Output.Size).To(Equal(size))
```

Comment:
```suggestion
			Expect(br.Status.Output.Digest).To(Equal(imageDigest))
			Expect(br.Status.Output.Size).To(Equal(230))
```

Again one more time in the same file.